src/main/java/com/aws/greengrass/componentmanager/plugins/docker/DefaultDockerClient.java

```diff
@@ -150,6 +152,42 @@ public void pullImage(Image image) throws DockerServiceUnavailableException, Inv
         }
     }
 
+    /**
+     * Check if an image exists locally.
+     *
+     * @param image image to check locally
+     * @throws DockerServiceUnavailableException   an error that can be potentially fixed through retries
+     * @throws UserNotAuthorizedForDockerException when current user is not authorized to use docker
+     * @throws DockerImageQueryException unexpected error
+     */
+    public boolean imageExistsLocally(Image image) throws DockerServiceUnavailableException,
+            UserNotAuthorizedForDockerException, DockerImageQueryException {
+        CliResponse response = runDockerCmd(String.format("docker images -q %s", image.getImageFullName()));
```

Comment:
Does the full name include the image version? i.e. tag or digest? We'd need that if we want to check if the specific version exists locally because a new component version could be attempting to download a new docker image version. Also, what happens when the artifact uri doesn't include a tag or digest (suggesting the latest version is desired)? We should always download in that case because the local version might be stale